src/analysis/normed_space/compact_operator.lean

```diff
@@ -0,0 +1,375 @@
+maps. Instead we define it as a predicate over bare functions, although it really only makes sense
+for linear functions, because Lean is really good at finding coercions to bare functions (whereas
+coercing from continuous linear maps to linear maps often needs type ascriptions).
+
+## TODO
+
+Once we have the strong operator topology on spaces of linear maps between to TVSs,
+`is_closed_set_of_is_compact_operator` should be generalized to this setup.
+
+## References
+
+* Bourbaki, *Spectral Theory*, chapters 3 to 5, to be published (2022)
+
+## Tags
+
+Compact operator
+-/
+
+open function set filter bornology metric
+
+open_locale pointwise big_operators topological_space
+
+/-- A compact operator between two TVSs. This definition is usually given as "there exists a
+neighborhood of zero whose image is contained in a compact set", but we choose a definition which
+involves fewer existential quantifiers and replaces images with preimages.
+
+We prove the equivalence in `is_compact_operator_iff_exists_mem_nhds_image_subset_compact`. -/
+def is_compact_operator {M₁ M₂ : Type*} [has_zero M₁] [topological_space M₁]
+  [topological_space M₂] (f : M₁ → M₂) : Prop :=
+∃ K, is_compact K ∧ f ⁻¹' K ∈ (𝓝 0 : filter M₁)
+
+lemma is_compact_operator_zero {M₁ M₂ : Type*} [has_zero M₁] [topological_space M₁]
+  [topological_space M₂] [has_zero M₂] : is_compact_operator (0 : M₁ → M₂) :=
+⟨{0}, is_compact_singleton, mem_of_superset univ_mem (λ x _, rfl)⟩
+
+section characterizations
+
+section
+
+variables {R₁ R₂ : Type*} [semiring R₁] [semiring R₂] {σ₁₂ : R₁ →+* R₂} {M₁ M₂ : Type*}
+  [topological_space M₁] [add_comm_monoid M₁] [topological_space M₂]
+
+lemma is_compact_operator_iff_exists_mem_nhds_image_subset_compact (f : M₁ → M₂) :
+  is_compact_operator f ↔ ∃ V ∈ (𝓝 0 : filter M₁), ∃ (K : set M₂), is_compact K ∧ f '' V ⊆ K :=
+⟨λ ⟨K, hK, hKf⟩, ⟨f ⁻¹' K, hKf, K, hK, image_preimage_subset _ _⟩,
+ λ ⟨V, hV, K, hK, hVK⟩, ⟨K, hK, mem_of_superset hV (image_subset_iff.mp hVK)⟩⟩
+
+lemma is_compact_operator_iff_exists_mem_nhds_is_compact_closure_image [t2_space M₂]
+  (f : M₁ → M₂) :
+  is_compact_operator f ↔ ∃ V ∈ (𝓝 0 : filter M₁), is_compact (closure $ f '' V) :=
+begin
+  rw is_compact_operator_iff_exists_mem_nhds_image_subset_compact,
+  exact ⟨λ ⟨V, hV, K, hK, hKV⟩, ⟨V, hV, compact_closure_of_subset_compact hK hKV⟩,
+         λ ⟨V, hV, hVc⟩, ⟨V, hV, closure (f '' V), hVc, subset_closure⟩⟩,
+end
+
+end
+
+section bounded
+
+variables {𝕜₁ 𝕜₂ : Type*} [nontrivially_normed_field 𝕜₁] [semi_normed_ring 𝕜₂] {σ₁₂ : 𝕜₁ →+* 𝕜₂}
+  {M₁ M₂ : Type*} [topological_space M₁] [add_comm_monoid M₁] [topological_space M₂]
+  [add_comm_monoid M₂] [module 𝕜₁ M₁] [module 𝕜₂ M₂] [has_continuous_const_smul 𝕜₂ M₂]
+
+lemma is_compact_operator.image_subset_compact_of_vonN_bounded {f : M₁ →ₛₗ[σ₁₂] M₂}
+  (hf : is_compact_operator f) {S : set M₁} (hS : is_vonN_bounded 𝕜₁ S) :
+  ∃ (K : set M₂), is_compact K ∧ f '' S ⊆ K :=
```

Comment:
Could you please later re-state this lemma and the next in terms of the metric notion of boundedness?